optuna/visualization/_contour.py

```diff
@@ -218,12 +220,12 @@ def _get_contour_subplot(
                 infeasible.x.append(x_value)
                 infeasible.y.append(y_value)
 
-    z_values = [
-        [float("nan") for _ in range(len(info.xaxis.indices))]
-        for _ in range(len(info.yaxis.indices))
-    ]
-    for (x_i, y_i), z_value in info.z_values.items():
-        z_values[y_i][x_i] = z_value
+    z_values = np.full((len(y_indices), len(x_indices)), np.nan)
+
+    xys = np.array(list(info.z_values.keys()))
+    zs = np.array(list(info.z_values.values()))
+
+    z_values[xys[:, 0], xys[:, 1]] = zs
```

Comment:
Sorry, I missed the change here:
```suggestion
    z_values[xys[:, 1], xys[:, 0]] = zs
```